pre_commit_hooks/insert_license.py

```diff
@@ -274,22 +280,22 @@ def try_update_year_range(
     :return: source file contents and a flag indicating update
     """
     current_year = datetime.now().year
-    for i in range(license_header_index, len(src_file_content)):
+    for i in range(license_header_index, license_header_index + license_lines):
         line = src_file_content[i]
         matches = _YEAR_RANGE_PATTERN.findall(line)
         if matches:
             match = matches[-1]
             start_year = int(match[:4])
-            end_year = match[5:]
-            if not end_year or int(end_year) < current_year:
+            end_year = match[5:].lstrip(" -,")
+            if not end_year and start_year < current_year or end_year and int(end_year) < current_year:
                 updated = line.replace(match,
-                                    str(start_year) + '-' + str(current_year))
+                                       str(start_year) + '-' + str(current_year))
                 # verify the current list of years ends in the current one
                 if _YEARS_PATTERN.findall(updated)[-1][-4:] != str(current_year):
-                    print(f"Unable to update year range in line: {line.rstrip()}. Got: {updated.rstrip()}")
-                    break
+                    raise RuntimeError(f"Unable to update year range in line: {line.rstrip()}. Got: {updated.rstrip()}")
```

Comment:
```suggestion
                    raise RuntimeError(f"Year range detected in license header, but we were unable to update it on line: {line.rstrip()}. Got: {updated.rstrip()}")
```
Also, have you tested if the name of the file is mentioned in the final error mesage reported by `pre-commit`?
That would be very useful :)

Reply:
Good idea. I added a mechanism to capture and inspect stdout.